packages/nimble-components/src/table/specs/table-column-specs/table-column-file-size.md

```diff
@@ -0,0 +1,294 @@
+# File Size Table Column
+
+## Overview
+
+The file size table column will provide a way to visualize numeric data that represents a file size (or other binary data size). It will automatically convert the raw value to use an appropriate display unit.
+
+### Background
+
+[Work Item](https://ni.visualstudio.com/DevCentral/_workitems/edit/2526114)
+
+### Non-goals
+
+-   Configurable number of decimal places
+-   Configurable units (e.g. display all values in KB)
+-   Grouping by value ranges/buckets
+
+### Features
+
+-   Converts the byte value to largest unit (e.g. KB, MB, GB) that results in a value of >=1
+-   Option to use 1024-based units (e.g. KiB) or 1000-based units (e.g. KB)
+-   Displays up to one decimal place
+-   Sorts/groups by numeric value (number of bytes) rather than display string
+-   Localized number and unit strings
+
+### Risks and Challenges
+
+### Prior Art/Examples
+
+Size column in [SLE Files grid](https://dev.lifecyclesolutions.ni.com/files/)
+
+Windows explorer (details view)
+
+---
+
+## Design
+
+Below is an example of how the `nimble-table-column-file-size` would be used within a `nimble-table`:
+
+```HTML
+<nimble-table>
+    <nimble-table-column-file-size field-name="size">Size</nimble-table-column-file-size>
+</nimble-table>
+```
+
+### API
+
+`nimble-table-column-file-size`
+
+-   `field-name` - name of the record field containing the data. Data must be a `number` byte count.
+-   `unit-type` - "binary" (KiB, MiB, GiB, TiB, PiB) or "decimal" (KB, MB, GB, TB, PB); defaults to "decimal"
+
+The component will extend `TableColumnTextBase`, thereby including the APIs for a **groupable** and **fractional-width** column.
+
+### Anatomy
+
+### Angular integration
+
+An Angular directive will be created for the component. The component will not have form association, so a `ControlValueAccessor` will not be created.
+
+### Blazor integration
+
+A Blazor wrapper will be created for the component.
+
+### Visual Appearance
+
+Will use same text styling as the `nimble-table-column-text`.
+
+## Implementation
+
+```ts
+class TableColumnFileSize extends TableColumnTextBase {
```

Comment:
Much of the implementation included here doesn't seem to pertain to the unique aspects of this column type. I would vote to remove any typical/boilerplate-ish code. In fact, I'm not sure there's much value in even providing the snippet in addition to the API description above. Most of the other implementation here is just a mirror of other patterns we have for dealing with how to update internal state.

Reply:
I've removed most of the code snippets, except for the implementation of the conversion/formatting function. I added a couple of examples of markup and rendered output, as you suggested.